src/Decoder.c

```diff
@@ -4987,19 +4986,14 @@ ZyanStatus ZydisDecoderDecodeFull(const ZydisDecoder* decoder,
 
     ZydisDecoderContext context;
     ZYAN_CHECK(ZydisDecoderDecodeInstruction(decoder, &context, buffer, length, instruction));
+    ZYAN_CHECK(ZydisDecoderDecodeOperands(decoder, &context, instruction, operands,
+        instruction->operand_count));
 
-    const ZyanBool visible_only = flags & ZYDIS_DFLAG_VISIBLE_OPERANDS_ONLY;
-    const ZyanU8 count = visible_only
-        ? instruction->operand_count_visible
-        : instruction->operand_count;
+    ZYAN_ASSERT(instruction->operand_count <= instruction->operand_count);
+    ZYAN_MEMSET(&operands[instruction->operand_count], 0,
+        (ZYDIS_MAX_OPERAND_COUNT - instruction->operand_count) * sizeof(operands[0]));
```

Comment:
I figured that it'd be rather unintuitive for this function not to zero the remaining entries now that it expects an array of a fixed size, but I'm open to changing this.

Reply:
I think we should keep it like this. It's consistent with the "simple disassembler API" where the complete struct is nulled. If performance is a concern, we can refer to the advanced functions as well.